app/global_namespace.go

```diff
@@ -3,34 +3,44 @@ package app
 import (
 	"fmt"
 
-	"github.com/rancher/rancher/pkg/settings"
-
 	"github.com/rancher/rancher/pkg/namespace"
+	"github.com/rancher/rancher/pkg/settings"
 	"github.com/rancher/types/config"
 	"github.com/sirupsen/logrus"
 
 	corev1 "k8s.io/api/core/v1"
 	k8serrors "k8s.io/apimachinery/pkg/api/errors"
 )
 
-func addCattleGlobalNamespace(management *config.ManagementContext) error {
+func addCattleGlobalNamespaces(management *config.ManagementContext) error {
+	if _, err := createNamespace(namespace.GlobalNamespace, management); err != nil {
+		return err
+	}
+
+	if _, err := createNamespace(namespace.NodeTemplateGlobalNamespace, management); err != nil {
+		return err
+	}
+
+	logrus.Debugf("calling sync for driver metadata")
+	management.Management.Settings("").Controller().Enqueue("", settings.RkeMetadataConfig.Name)
+
+	return nil
+}
 
+func createNamespace(namespace string, management *config.ManagementContext) (*corev1.Namespace, error) {
```

Comment:
Don't return the namespace, it isn't going to work anyways since the create call is eating the ns.